R/module_nested_tabs.R

```diff
@@ -172,18 +172,16 @@ srv_nested_tabs.teal_modules <- function(id, datasets, modules, is_module_specif
     logger::log_trace("srv_nested_tabs.teal_modules initializing the module { deparse1(modules$label) }.")
 
     labels <- vapply(modules$children, `[[`, character(1), "label")
-    modules_reactive <- sapply(
-      names(modules$children),
-      function(module_id) {
-        srv_nested_tabs(
-          id = module_id,
-          datasets = datasets[[labels[module_id]]],
-          modules = modules$children[[module_id]],
-          is_module_specific = is_module_specific,
-          reporter = reporter
-        )
-      }
-    )
+    modules_reactive <- lapply(names(modules$children), function(module_id) {
+      srv_nested_tabs(
+        id = module_id,
+        datasets = datasets[[labels[module_id]]],
+        modules = modules$children[[module_id]],
+        is_module_specific = is_module_specific,
+        reporter = reporter
+      )
+    })
+    names(modules_reactive) <- names(modules$children)
```

Comment:
What is wrong with `sapply` in this case? `sapply` adds names for free. In my opinion the only controversy with `sapply` could be `simplify = TRUE` (default), but we can just use `simplify = FALSE`.